src/mono/wasm/runtime/jiterpreter-jit-call.ts

```diff
@@ -830,7 +791,10 @@ function generate_wasm_body(
 
     // If the call threw a JS or wasm exception, set the thrown flag
     if (builder.options.enableWasmEh) {
-        builder.appendU8(WasmOpcode.catch_all);
+        builder.appendU8(WasmOpcode.catch_);
+        builder.appendULeb(builder.getTypeIndex("__cpp_exception"));
+        builder.callImport("begin_catch");
```

Comment:
the JS code in `mono_interp_invoke_wasm_jit_call_trampoline` has more branches than this. Could you please explain what is the difference ?

Reply:
Inside this branch we know wasm EH is enabled, so we by definition will not encounter a JS exception... I think. Even if we did, we couldn't handle it natively using wasm code - we would have to use JS.

invoke_wasm_jit_call_trampoline on the other hand needs to work whether wasm EH is enabled or disabled. It is responsible for handling JS exceptions on behalf of trampolines (which can't handle them).